### PR TITLE
[7723]Fix user's email is not unique

### LINKED ISSUE #7723: user's email is not unique
## CKAN version
2.10.1

## Describe the bug
User's email field is case-sensitive, therefore user with one email address can be added multiple times. 

### Steps to reproduce
```
ckan user add user1 email=abc@local password=12345678
ckan user add user2 email=ABC@local password=12345678
```

### Expected behavior
second command should fail with message about email already exists.

### PR DESCRIPTION
**Pull request by TomeCirun**
Fixes #7723

### Features:

- [x] includes tests covering changes
- [ ] includes updated documentation
- [ ] includes user-visible changes
- [ ] includes API changes
- [ ] includes bugfix for possible backport

Please [X] all the boxes above that apply

### CODE REVIEW

File: ckan/logic/validators.py [hunk 1]
```diff
@@ -995,7 +995,7 @@ def email_validator(value: Any, context: Context) -> Any:
     if value:
         if not email_pattern.match(value):
             raise Invalid(_('Email {email} is not a valid format').format(email=value))
-    return value
+    return value.lower()
```
Comment (amercader) [inline ckan/logic/validators.py:998]: @TomeCirun last time this was discussed we agreed to not change the user input (eg, if users input `ABC@example.com` we don't want to store `abc@example.com` in the database). Your changes here will force lower case in all incoming emails.

I don't know the best alternative off the top of my head, I don't have enough context as to why we dropped the `email_is_unique` validator in https://github.com/ckan/ckan/pull/6916 in favour of a db-based check but we need to give this further thought

**Comment by Labuschagne-Miro** [comment]
@TomeCirun are you still working on this pull request? Are you able to fix it with another solution?

File: ckan/logic/validators.py (hunk 1 above)
Comment (TomeCirun) [reply ckan/logic/validators.py]: Hey @amercader ,

I recall why we removed the `email_is_unique` validator in **#6916**. The concern was that concurrent `user_update` or `user_create` requests with the same email might lead to duplicated entries.

As highlighted by Ian in the original [issue](https://github.com/ckan/ckan/issues/6893#issuecomment-1148684210), we opted to ensure unique active emails by implementing a composite constraint at the database level (which I implemented in the PR). We believed this database-level constraint alone would be sufficient, leading to the removal of the `email_is_unique` validator.

However, there's no harm in reintroducing `email_is_unique`. I've made some minor adjustments to address our requirements.

please let me know your opinion on this

**Comment by Labuschagne-Miro** [comment]
@TomeCirun , https://github.com/ckan/ckan/pull/8016

**Comment by Vishal-NEC** [comment]
> @TomeCirun , #8016

@TomeCirun Really sorry for the confusion made, Earlier I am assuming that You left this issue, Hence I started Fixing on that
I have already shared a Patch for review 2 weeks ago. so that I Raise the PR.

**Comment by Labuschagne-Miro** [comment]
@Vishal-NEC  you should first confirm without working any issue or pull request. Assumption are not expected from a developer. Please follow community guidelines properly otherwise your I'd will be blocked.

**Comment by TomeCirun** [comment]
Hey @amercader ,  is there anything else I can do in this PR?

**Comment by amercader** [comment]
Sorry this got forgotten @TomeCirun , this looks good, the only thing missing is a test in `tests/logic/create.py`:

1. Create a user with "some_email@example.org"
2. When you try to create a new user with "Some_Email@example.org" it fails with a ValidationError

**Comment by TomeCirun** [comment]
@amercader done, thanks for you help

**Comment by TomeCirun** [comment]
Hey @amercader polite reminder :)

**Comment by amercader** [comment]
Thanls @TomeCirun , this is great, sorry it took a while

**Pull request merged**